sympy/combinatorics/graycode.py

```diff
@@ -85,6 +86,7 @@ def __new__(cls, n, *args, **kw_args):
                 'Gray code dimension must be a positive integer, not %i' % n)
         n = int(n)
         args = (n,) + args
+        args = (_sympify(arg) for arg in args)
```

Comment:
This makes me worry that any calcs that assume args are ints will now be slowed down a lot. This was something we had to be careful of in the ntheory/prime work.

Reply:
To be clear I don't intend for this to be merged.